pkg/es/config/config.go

```diff
@@ -45,7 +45,7 @@ import (
 
 // Configuration describes the configuration properties needed to connect to an ElasticSearch cluster
 type Configuration struct {
-	Servers                        []string       `mapstructure:"server_urls"`
+	Servers                        []string       `mapstructure:"server_urls" validate:"required,min=1,dive,url"`
```

Comment:
I wonder if there's a test we could write for ensuring that these annotations are correct (btw, what is "dive"?)

We could do it brute-force and just write individual tests for individual fields, but it seems redundant as that would be already done in the validator package (hopefully). But at the same time, having no tests to even ensure that the syntax of the annotations is correct is troublesome.

Reply:
Comment for L58 and below - can you try removing `yaml:...` annotations? I think they are left-overs from earlier attempts to build on OTEL Coll, I don't think we're using them anywhere.

Reply:
>Comment for L58 and below - can you try removing yaml:... annotations? I think they are left-overs from earlier attempts to build on OTEL Coll, I don't think we're using them anywhere.

removed it

Reply:
> (btw, what is "dive"?)

dive instructs the validation library to examine each element within the slice individually

Reply:
> I wonder if there's a test we could write for ensuring that these annotations are correct 
> 
> We could do it brute-force and just write individual tests for individual fields, but it seems redundant as that would be already done in the validator package (hopefully). But at the same time, having no tests to even ensure that the syntax of the annotations is correct is troublesome.

we can write unit tests that check whether the validation behaves as expected. Instead of testing individual fields, we can create a set of test cases that cover various scenarios, including both valid and invalid configurations.
should i create one?

Reply:
but that wouldn't be what I am after, such test would simply verify that the Validator is working as expected, on select fields. But if I declare a field with annotations that don't make sense, I want to catch that in unit tests. E.g. misspell `dive` as `diev` (btw weird option, "recursive" would've been much more self explanatory).

Reply:
We can use the "[github.com/go-playground/validator](https://github.com/go-playground/validator)" library in our unit tests to detect misspelled annotations and fields with invalid annotation combinations.
```go 
package main

import (
	"github.com/go-playground/validator"
)

type Webhook struct {
	CallbackURL string `validate:"url,badAnnotation"` // badAnnotation is not a valid tag
}

func main() {
	webhook := Webhook{
		CallbackURL: "https://www.example.com/",
	}
	validate := validator.New()
	validate.Struct(webhook)

}
```
```
>output 
panic: Undefined validation function 'badAnnotation' on field 'CallbackURL'

goroutine 1 [running]:
github.com/go-playground/validator.(*Validate).parseFieldTagsRecursive(0xc00002e240, {0x526987?, 0x55812d?}, {0x526971, 0xb}, {0x0, 0x0}, 0x0)
        /home/harsh/go/pkg/mod/github.com/go-playground/validator@v9.31.0+incompatible/cache.go:289 +0x9d9
github.com/go-playground/validator.(*Validate).extractStructCache(0xc00002e240, {0x538520?, 0xc00020a950?, 0x7fda1aac3198?}, {0x52160c, 0x7})
        /home/harsh/go/pkg/mod/github.com/go-playground/validator@v9.31.0+incompatible/cache.go:150 +0x4ec
github.com/go-playground/validator.(*validate).validateStruct(0xc000118480, {0x58b0c0, 0x68faa0}, {0x538520?, 0xc00020a950?, 0xc000242000?}, {0x538520?, 0xc00020a950?, 0x40e3a5?}, {0x58cd60, ...}, ...)
        /home/harsh/go/pkg/mod/github.com/go-playground/validator@v9.31.0+incompatible/validator.go:37 +0x196
github.com/go-playground/validator.(*Validate).StructCtx(0xc00002e240, {0x58b0c0, 0x68faa0}, {0x538520?, 0xc00020a950?})
        /home/harsh/go/pkg/mod/github.com/go-playground/validator@v9.31.0+incompatible/validator_instance.go:304 +0x491
github.com/go-playground/validator.(*Validate).Struct(...)
        /home/harsh/go/pkg/mod/github.com/go-playground/validator@v9.31.0+incompatible/validator_instance.go:277
main.main()
        /home/harsh/Testing/main.go:16 +0x65
exit status 2

Reply:
what about "github.com/asaskevich/govalidator", will it do the same? We do not use "github.com/go-playground/validator"

Reply:
>will it do the same?

Nope, it doesn't do the same. I actually tried using the existing validator at first.